src/transformers/generation/configuration_utils.py

```diff
@@ -298,6 +298,9 @@ def __init__(self, **kwargs):
         self.validate()
 
     def __eq__(self, other):
+        if not isinstance(other, GenerationConfig):
+            return False
+
```

Comment:
This is a redundant check, but `torch.compile()` is throwing an exception without it 🤷

Reply:
@gante Would you be able to send a repro for the issue that occurs when you remove this line? There's interest in fixing this on the pytorch side https://github.com/pytorch/pytorch/issues/96205.

Reply:
This is only for generation config.

This is only for pretrained config:

https://github.com/pytorch/torchdynamo/commit/e88bd663a0d6e38d4fa0fd3b737f8311e11585ec#diff-fee9e700f4a2e99d7072c1cb178d221517a0ac62aca2c39c24211a284cfec4c2R76

Maybe we should find a general fix for all *Config?